src/data_conn.c

```diff
@@ -64,13 +64,27 @@ int replica_timeout = REPLICA_DEFAULT_TIMEOUT;
 		rcomm_cmd->resp_list[idx].io_resp_hdr.status =		\
 		    ZVOL_OP_STATUS_FAILED;				\
 		rcomm_cmd->resp_list[idx].data_ptr = NULL;		\
+		/*							\
+		 * cleanup_deadlist thread performs cleanup of 		\
+		 * rcomm_cmd. if the response from all replica is 	\
+		 * received. Since we are avoiding locking for 		\
+		 * rcomm_cmd, we will update response status in 	\
+		 * rcomm_cmd at last.					\
+		 */							\
 		if (rcomm_cmd->state != CMD_EXECUTION_DONE) {		\
 			rcomm_cmd->resp_list[idx].status |= 		\
 			    RECEIVED_ERR;				\
 			pthread_cond_signal(_cond);			\
-		} else							\
+		} else {						\
 			rcomm_cmd->resp_list[idx].status |= 		\
 			    RECEIVED_ERR;				\
+			REPLICA_DEBUGLOG("error set for command(%lu)"	\
+			    " for replica(%lu)\n",			\
+			    rcomm_cmd->io_seq, r->zvol_guid);		\
```

Comment:
rcomm_cmd is again accessed after setting status, which can lead to crash

Reply:
ok